frontend/lib/requests/user-activity.ts

```
@@ -0,0 +1,499 @@
+      timestamp.getDate(),
+      timestamp.getHours(),
+      timestamp.getMinutes(),
+    ).toISOString();
+
+    const key = `${activity.type}-${roundedTime}-${JSON.stringify(activity.details?.dropletId || "")}`;
+
+    // Prefer enrollment data over PostHog for enrollments, completions, and ratings
+    if (!seen.has(key)) {
+      seen.set(key, activity);
+    } else {
+      const existing = seen.get(key)!;
```

ESLint

[error] 'existing' is assigned a value but never used.

(@typescript-eslint/no-unused-vars)